lib/parse.js

```diff
@@ -84,6 +135,35 @@ var INLINE_CODE = 'inlineCode';
 var BREAK = 'break';
 var ROOT = 'root';
 
+/** TODO */
+function isWhiteSpace(character) {
```

Comment:
> Im hoping this will also fix the safari issues!

I just ran the benchmark on JavaScriptCore and it showed that RegExp version (`/\s/.test`) is faster than `isWhiteSpace` (5 times faster, actually). The version of JSC is rather old though.

> I indeed hope the new "algorithmic" approach performs better, but mostly, that it will provide performs composability

That is true, and I agree that algorithmic approach is better than parsing with RegExps! But I think that simple one-character RegExps (such as `/\s/` or `/[0-9a-fA-F]/`) can still be used when they make the code simpler and more concise. There is nothing more "algorithmic" in

```js
if (
    char === '0' ||
    char === '1' ||
    char === '2' ||
// ...
```

 compared to `/[0-9a-fA-F]/`. It is just the same, except that when written in such an explicit form it is also prone to typos and missing cases.